adagucserverEC/CDBAdapterPostgreSQL.cpp

```diff
@@ -1080,3 +1084,86 @@ int CDBAdapterPostgreSQL::addFilesToDataBase() {
 #endif
   return 0;
 }
+
+int CDBAdapterPostgreSQL::storeLayerMetadata(const char *datasetName, const char *layerName, const char *metadataKey, const char *metadataBlob) {
+#ifdef MEASURETIME
+  StopWatch_Stop(">CDBAdapterPostgreSQL::storeLayerMetadata");
+#endif
+  CPGSQLDB *dataBaseConnection = getDataBaseConnection();
+  if (dataBaseConnection == NULL) {
+    return -1;
+  }
+
+  CT::string tableColumns("datasetname varchar (255), layername varchar (255), metadatakey varchar (255), updatetime TIMESTAMPTZ, blob JSONB, PRIMARY KEY (datasetname, layername, metadatakey)");
+
+  int status = dataBaseConnection->checkTable("layermetadata", tableColumns.c_str());
+  if (status == 1) {
+    CDBError("\nFAIL: Table layermetadata could not be created: %s", tableColumns.c_str());
+    throw(__LINE__);
+  }
+
+  CT::string updateTime = CTime::currentDateTime().c_str();
+  updateTime.setSize(19);
```

Comment:
What does this do? Convert through c_str(), and resive?

Reply:
Had to reduce the size because the postgresl timestamp column does not accept the trailing `Z`

Reply:
Ok. Another thing is that `CTime::currentDateTime()` already returns a CT::string, why go through `c_str()`?

Reply:
Did you check if the times in the table are correct if the DB and client are not running in UTC time zone?
Because I think inserting a time without timeonze into time `TIMESTAMPTZ` column might trigger timezone conversion.

See [here](https://www.postgresql.org/docs/current/datatype-datetime.html#DATATYPE-DATETIME-INPUT-TIME-STAMPS):
`For timestamp with time zone, the internally stored value is always in UTC (Universal Coordinated Time, traditionally known as Greenwich Mean Time, GMT). An input value that has an explicit time zone specified is converted to UTC using the appropriate offset for that time zone. If no time zone is stated in the input string, then it is assumed to be in the time zone indicated by the system's [TimeZone](https://www.postgresql.org/docs/current/runtime-config-client.html#GUC-TIMEZONE) parameter, and is converted to UTC using the offset for the timezone zone.`

Can we get the timestamp in a format like this to avoid the issue:
`1999-01-08 04:05:06 -8:00`

Maybe instead of resizing, just replace `Z` by ` 0:00`?

Reply:
I was not able to insert timestamps with the suggested format. But I changed the type from TIMESTAMPTZ to TIMESTAMP and now I see UTC timestamps in the table.